bundles/org.openhab.binding.freeathome/src/main/java/org/openhab/binding/freeathome/internal/handler/FreeAtHomeBridgeHandler.java

```diff
@@ -131,10 +136,86 @@ public Collection<Class<? extends ThingHandlerService>> getServices() {
         return List.of(FreeAtHomeDiscoveryService.class);
     }
 
+    /**
+     * Method to fetch SysApp Version
+     */
+    public boolean fetchSysapVersion() {
+        String url = baseUrl + "/rest/configuration";
+        try {
+            HttpClient client = httpClient;
+            Request req = client.newRequest(url);
+
+            if (req == null) {
+                logger.error("Invalid request object in fetchSysapVersion with the URL [ {} ]", url);
+                return false;
+            }
+
+            ContentResponse response = req.send();
+
+            if (response.getStatus() != 200) {
+                logger.error("HTTP request failed in fetchSysapVersion with status [{}] and reason [{}]",
+                        response.getStatus(), response.getReason());
+                return false;
+            }
+
+            String configString = new String(response.getContent());
+
+            JsonReader reader = new JsonReader(new StringReader(configString));
+            reader.setLenient(true);
+            JsonElement jsonTree = JsonParser.parseReader(reader);
+
+            if (!jsonTree.isJsonObject()) {
+                logger.error("Invalid jsonObject in fetchSysapVersion with the URL [ {} ]", url);
```

Comment:
Please also adjust other log levels. error should only be used for a system wide issue or serieus bug. Not if the binding can recover or the scope is limited to the binding.

```suggestion
                logger.warn("Invalid jsonObject in fetchSysapVersion with the URL [ {} ]", url);
```